### PR TITLE
[Enhance] Update HPA README

### PR DESCRIPTION
**Pull request by bryan-hz**
Sync the instructions from the latest HPA public doc: https://cloud.google.com/stackdriver/docs/managed-prometheus/hpa

cc.@courageJ

### CODE REVIEW

File: examples/hpa/README.md [hunk 1]
```diff
@@ -1,3 +1,169 @@
+# custom-metrics-stackdriver-adapter example instructions
```
Comment (lyanco) [inline examples/hpa/README.md:1]: At the top, add a link to the HPA docs: 

For further instructions, see https://cloud.google.com/stackdriver/docs/managed-prometheus/hpa

File: examples/hpa/README.md [hunk 2]
```diff
@@ -1,3 +1,169 @@
...
+       interval: 30s
+   ```
+
+5. Deploy the new PodMonitoring resource:
+
+   ```sh
+   kubectl -n default apply -f podmonitoring.yaml
+   ```
+
+   Within a couple of minutes, {{managed_prometheus_name_short}} processes the
```
Comment (lyanco) [inline examples/hpa/README.md:49]: All these {{variables}} need to be expanded out, as they're not defined here.

{{managed_prometheus_name_short}} == Managed Service for Prometheus
{{monitoring_name}} == Cloud Monitoring
{{metrics_explorer}} == Metrics Explorer

File: examples/hpa/README.md [hunk 3]
```diff
@@ -1,3 +1,169 @@
...
+      To verify the suffix, look up the metric in {{monitoring_name}} by using
+      {{metrics_explorer}}.
+
+6. Update the deployed HPA to query the metric from {{monitoring_name}}. The
+   metric `foo` is ingested as `prometheus.googleapis.com/foo/gauge`. To make
+   the metric queryable by the deployed HorizontalPodAutoscaler resource, you
+   use the long-form name in the deployed HPA, but you have to modify it
+   by replacing the all forward slashes (`/`)  with the pipe character (`|`):
+   `prometheus.googleapis.com|foo|gauge`. For more information, see the
+   [Metrics available from Stackdriver section][cm-sd-adapter-metrics]{:class="external"}
```
Comment (lyanco) [inline examples/hpa/README.md:64]: This link isn't showing

**Review by lyanco — CHANGES_REQUESTED** [review]

File: examples/hpa/README.md [hunk 4]
```diff
@@ -1,3 +1,169 @@
...
+2. Install Custom Metrics Stackdriver Adapter in your cluster.
+
+   ```sh
+   kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/deploy/production/adapter_new_resource_model.yaml
+   ```
+
+3. Deploy an example Prometheus metrics exporter and an HPA resource:
+
+   ```sh
+   kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/examples/prometheus-to-sd/custom-metrics-prometheus-sd.yaml
```
Comment (courageJ) [inline examples/hpa/README.md:20]: I think we discussed to use [https://www.google.com/url?sa=D&q=https%3A%2F%2Fgithub.com%2FGoogleCloudPlatform%2Fk8s-stackdriver%2Fblob%2Fmaster%2Fcustom-metrics-stackdriver-adapter%2Fexamples%2Fdirect-to-sd%2Fcustom-metrics-sd.yaml instead?](https://github.com/GoogleCloudPlatform/k8s-stackdriver/blob/master/custom-metrics-stackdriver-adapter/examples/direct-to-sd/custom-metrics-sd.yaml) instead?

Comment (lyanco) [reply examples/hpa/README.md:20]: That doesn't work as it emits the metric in k8s_container format, not GMP format

File: examples/hpa/README.md (hunk 1 above)
Comment (bryan-hz) [reply examples/hpa/README.md:3]: done

File: examples/hpa/README.md (hunk 2 above)
Comment (bryan-hz) [reply examples/hpa/README.md]: done

File: examples/hpa/README.md (hunk 3 above)
Comment (bryan-hz) [reply examples/hpa/README.md]: done

File: examples/hpa/README.md (hunk 4 above)
Comment (courageJ) [reply examples/hpa/README.md:22]: I think we could use https://raw.githubusercontent.com/GoogleCloudPlatform/prometheus-engine/v0.5.0/examples/example-app.yaml instead?

Comment (courageJ) [reply examples/hpa/README.md:22]: I am ok to use the current https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/examples/prometheus-to-sd/custom-metrics-prometheus-sd.yaml if it emits the expected metric but I think the example contains a redundant container prometheus-to-sd, which is not in use.

Comment (lyanco) [reply examples/hpa/README.md:22]: looks like `example-app` doesn't emit any static-valued metrics besides `up`, and those are most useful for this scenario. Can always change it to use example-app later I suppose.

**Review by lyanco — APPROVED** [review]

File: examples/hpa/README.md [hunk 5]
```diff
@@ -1,3 +1,171 @@
...
+[version v0.13.1 of the adapter](https://github.com/GoogleCloudPlatform/k8s-stackdriver/releases/tag/cm-sd-adapter-v0.13.1).
+
+To set up an example HPA configuration using the Custom Metrics Stackdriver
+Adapter, do the following:
+
+1. Set up Managed Service for Prometheus in your cluster.
+2. Install Custom Metrics Stackdriver Adapter in your cluster.
+
+   ```sh
+   kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/deploy/production/adapter_new_resource_model.yaml
```
Comment (pintohutch) [inline examples/hpa/README.md:16]: nit: use a [permalink](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files) to specify the commit, or better yet, the tag, of the version of the yaml to install. This avoids us having to refresh the doc later if the state of `master` ever changes (for example if that repository migrates to the `main` naming convention for the base branch).

Comment (bryan-hz) [reply examples/hpa/README.md]: done

**Comment by courageJ** [comment]
@pintohutch friendly ping?

**Review by pintohutch — APPROVED** [review]

**Pull request merged**